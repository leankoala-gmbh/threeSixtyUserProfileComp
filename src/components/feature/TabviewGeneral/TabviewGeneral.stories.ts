import type { Meta, StoryObj } from '@storybook/vue3'
import TabviewGeneral from './TabviewGeneral.vue'

const meta: Meta<typeof TabviewGeneral> = {
  title: 'Components/Feature/TabviewGeneral',
  component: TabviewGeneral,
  tags: ['autodocs'],
  argTypes: {
    // Define your interactive Props here
    // example: {
    //  options: ['text', 'password', 'email', 'number'],
    //  control: {
    //    type: 'select'
    //  }
  },
  parameters: {
    docs: {
      description: {
        component: 'Some component _markdown_'
      }
    }
  }
}

export default meta
type Story = StoryObj<typeof TabviewGeneral>

// First Story of the Component
export const TabviewGeneralStory: Story = {
  // name: 'Self defined Story Name',
  render: (args) => ({
    components: { TabviewGeneral },
    setup() {
      return { args }
    },
    template: '<TabviewGeneral v-bind="args" />'
  }),
  args: {
    // Define the Data for your props
    // example: 'Prefilled value'
  }
}
