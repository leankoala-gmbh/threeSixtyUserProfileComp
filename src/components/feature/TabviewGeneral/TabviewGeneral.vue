<script lang="ts" setup>
import ProfileInfos from '@/components/feature/ProfileInfos/ProfileInfos.vue'
import Timezone from '@/components/feature/Timezone/Timezone.vue'
import { translator } from '@/composables/translator'
import { IProfileUser } from '@/types/general.interfaces'

const props = defineProps({
  userDataObj: {
    type: Object as () => IProfileUser,
    required: true
  },
  inactiveFieldsArr: {
    type: Array as () => string[],
    default: () => ['removeAccount']
  }
})

const updateTimezone = (event: string) => {
  window.mitt.emit('tsxUserProfile', {
    timezone: event
  })
}
</script>

<template>
  <div class="font-medium mb-4 text-lg">
    {{ translator('Sentence for general Tab') }}
  </div>
  <ProfileInfos
    :user-data="userDataObj"
    :inactive-fields="inactiveFieldsArr"
  />
  <Timezone
    v-if="!inactiveFieldsArr.includes('timezone')"
    :user-data="userDataObj"
    @update-timezone="updateTimezone"
  />
</template>
