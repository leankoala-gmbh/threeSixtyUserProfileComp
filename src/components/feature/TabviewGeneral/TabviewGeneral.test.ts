import { shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import TabviewGeneral from './TabviewGeneral.vue'

describe('TabviewGeneral.vue', () => {
  it('renders props.msg when passed', () => {
    const wrapper = shallowMount(TabviewGeneral, {
      props: {
        userDataObj: {
          name: 'test',
          email: 'features@leankoala.com'
        },
        inactiveFieldsArr: []
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
