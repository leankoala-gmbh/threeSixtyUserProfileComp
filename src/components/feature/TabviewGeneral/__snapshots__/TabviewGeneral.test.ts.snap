// Vitest Snapshot v1

exports[`TabviewGeneral.vue > renders props.msg when passed 1`] = `
"<div class=\\"font-medium mb-4 text-lg\\">A general sentence for general tab</div>
<profile-infos-stub userdata=\\"[object Object]\\" inactivefields=\\"\\"></profile-infos-stub>
<timezone-stub userdata=\\"[object Object]\\"></timezone-stub>"
`;
